packages/visx-xychart/src/components/series/BarGroup.tsx

```diff
@@ -0,0 +1,171 @@
+    height,
+  } = (useContext(DataContext) as unknown) as DataContextType<XScale, YScale, Datum>;
+
+  const barSeriesChildren = useMemo(
+    () =>
+      React.Children.toArray(children).filter(child =>
+        isChildWithProps<BarSeriesProps<XScale, YScale, Datum>>(child),
+      ),
+    [children],
+  ) as React.ReactElement<BarSeriesProps<XScale, YScale, Datum>>[];
+
+  // extract data keys from child series
+  const dataKeys: string[] = useMemo(
+    () => barSeriesChildren.map(child => child.props.dataKey ?? '').filter(key => key),
+    [barSeriesChildren],
+  );
+
+  // register all child data
+  useEffect(() => {
+    const dataToRegister = barSeriesChildren.map(child => {
+      const { dataKey: key, data, xAccessor, yAccessor } = child.props;
+      return { key, data, xAccessor, yAccessor };
+    });
+
+    registerData(dataToRegister);
+    return () => unregisterData(dataKeys);
+  }, [registerData, unregisterData, barSeriesChildren, dataKeys]);
+
+  // create group scale
+  const groupScale = useMemo(
+    () =>
+      scaleBand<string>({
+        domain: [...dataKeys], // @TODO order
+        range: [0, getScaleBandwidth(horizontal ? yScale : xScale)],
+        padding,
+      }),
+    [dataKeys, xScale, yScale, horizontal, padding],
+  );
+
+  const { showTooltip, hideTooltip } = useContext(TooltipContext) ?? {};
+  const handleMouseMove = useCallback(
+    (params?: HandlerParams) => {
+      const { svgPoint } = params || {};
+      // invoke showTooltip for each key so all data is available in context,
+      // and let Tooltip find the nearest point among them
+      dataKeys.forEach(key => {
+        const entry = dataRegistry.get(key);
+
+        if (entry && svgPoint && width && height && showTooltip) {
+          const datum = (horizontal ? findNearestDatumY : findNearestDatumX)({
+            point: svgPoint,
+            data: entry.data,
+            xScale,
+            yScale,
+            xAccessor: entry.xAccessor,
+            yAccessor: entry.yAccessor,
+            width,
+            height,
+          });
+          if (datum) {
+            showTooltip({
+              key,
+              ...datum,
+              svgPoint,
+            });
+          }
+        }
+      });
+    },
+    [dataKeys, dataRegistry, horizontal, xScale, yScale, width, height, showTooltip],
+  );
+  useEventEmitter('mousemove', handleMouseMove);
+  useEventEmitter('mouseout', hideTooltip);
+
+  const registryEntries = dataKeys.map(key => dataRegistry.get(key));
+
+  // if scales and data are not available in the registry, bail
+  if (registryEntries.some(entry => entry == null) || !xScale || !yScale || !colorScale) {
+    return null;
+  }
+
+  const [xMin, xMax] = xScale.range().map(Number);
+  const [yMax, yMin] = yScale.range().map(Number);
+
+  // try to figure out the 0 baseline for correct rendering of negative values
+  // we aren't sure if these are numeric scales or not ahead of time
+  const maybeXZero = xScale(0) ?? 0;
```

Comment:
some of this is duplicated in `BarSeries`. I'm going to focus on animated bars next, and will try to remove some of the duplication.